src/components/PDFView/index.native.js

```diff
@@ -1,50 +1,17 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {TouchableWithoutFeedback} from 'react-native';
-import PDF from 'react-native-pdf';
-import styles from '../../styles/styles';
-import * as StyleUtils from '../../styles/StyleUtils';
-import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
-import FullScreenLoadingIndicator from '../FullscreenLoadingIndicator';
-
-const propTypes = {
-    /** URL to full-sized image */
-    sourceURL: PropTypes.string,
-
-    /** Any additional styles to apply */
-    // eslint-disable-next-line react/forbid-prop-types
-    style: PropTypes.any,
-
-    ...windowDimensionsPropTypes,
-};
-
-const defaultProps = {
-    sourceURL: '',
-    style: {},
-};
-
-/**
- * On the native layer, we use a pdf library to display PDFs
- *
- * @param props
- * @returns {JSX.Element}
- */
+import BasePDFViewNative from './BasePDFViewNative';
```

Comment:
Should we rename this file from `index.native.js` to `index.ios.js`? Others might ignore the `index.android.js` file thinking that there is a native file which generally used for both android and iOS?

Reply:
Thanks for testing everything, @parasharrajat.

### Bouncing form
I'll fix that.

### Autofill bug
The input has `autocomplete="off"`. Is it possible that the browser stored information for this page during some previous testing - before I disabled autocomplete? If there is a saved password (for example on Chrome at `chrome://settings/passwords`), what happens if you delete it? 

### Android send button visibility
I'll fix that.

### Rename index.native.js
I'll rename it.

Reply:
Regarding the send button on Android... 

The input field doesn't lose focus on Android when the keyboard is closed directly. So it appears that I must use `withKeyboardState` to make things work correctly on Android native.

So I may as well refactor everything and move all this code into `AttachmentModal.js` as you suggested a couple weeks ago, @parasharrajat.

Assuming this all works as expected I'll need to make the following changes to `AttachmentModal.js`:

1. Add `withKeyboardState` to set of composed libraries.
2. Add `componentDidMount()` handler.
3. In `componentDidMount()` record initial screen height in state, and add `visualViewport` listener.
4. Add `componentWillUnmount()` handler to unsubscribe `visualViewport` listener.
5. Create `visualViewport` listener that (for small screens) updates `state.shouldShowSendButton` according to initial screen height and current screen height.
6. Add `componentDidUpdate()` handler that updates `state.shouldShowSendButton` according to the `props.isShown` property provided by `withKeyboardState`. This is specifically for native.
7. Add a file-type check to send button updates to ensure that the send button manipulation only occurs for PDF attachments. This is required to avoid potential regression issues when viewing other types of attachments.
8. Refactor `AttachmentModal.js` into a hybrid component with an `index.js`, `index.native.js` and `BaseAttachementModal.js`.

Shall I go ahead and do this, @parasharrajat?

Thanks again for testing everything. And sorry for all the hassle.

Reply:
AutoComplete is off though, this is weird. There are some ideas https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion. 

Like try name but it didn't work for me.

Reply:
Sorry I forgot to ask. What browser are you using?

Reply:
> Regarding the send button on Android... 

I see that the same issue also exists on Chrome Android. 

As per the mocks, it seems that send button is not visible on mobile devices when the keyboard is open. We will also have to worry about the tablet's landscape mode. That Send button really does not look good because of the small area. 

Also, will it require a Scrollview in case the screen has very less space? the form should not be cut off?

> Assuming this all works as expected I'll need to make the following changes to AttachmentModal.js:

About this plan, I am not too sure about the challenges. But looks good to me. Also please try the app on Android tablets to analyze if this will work there. 

With the current focus-based solution, there is a delay before the send button hides.

Let me know if you feel that this might not work and if you have better suggestions. Even if you have ideas where we don't have to do it.

Reply:
1. I don't have a solution to the autocomplete issue. You're seeing different behavior on your machine than I'm seeing on mine (macOS). Could you perhaps try making some minor code changes yourself to see if anything works? Note that the page you referenced states that

> For this reason, many modern browsers do not support autocomplete="off" for login fields.

2. I'll hide the send button whenever the keyboard is open on any device. And I'll verify that it works on Android tablets.

3. I'll put the password form in a ScrollView.

4. I don't think it's possible to resolve the delay before hiding the send button. This happens with all solutions on Android on both native and web. The problem is that the keyboard opens first, and then events are fired. Note that in `withKeyboardState.js` the event is called `keyboardDidShow`. So it fires **after** the keyboard is visible. There is in fact a `keyboardWillShow` event, but it doesn't work on Android: https://reactnative.dev/docs/keyboard

Reply:
> There is in fact a keyboardWillShow event, but it doesn't work on Android

yup, it won't work on our app.